src/uhs/transaction/wallet.cpp

```diff
@@ -25,22 +25,55 @@ namespace cbdc {
     auto transaction::wallet::mint_new_coins(const size_t n_outputs,
                                              const uint32_t output_val)
         -> transaction::full_tx {
-        transaction::full_tx ret;
+        transaction::full_tx tx;
+
+        const auto pubkey = generate_minter_key();
 
         for(size_t i = 0; i < n_outputs; i++) {
             transaction::output out;
-
-            const auto pubkey = generate_key();
-
             out.m_witness_program_commitment
                 = transaction::validation::get_p2pk_witness_commitment(pubkey);
-
             out.m_value = output_val;
+            tx.m_outputs.push_back(out);
+        }
 
-            ret.m_outputs.push_back(out);
+        const auto sighash = transaction::tx_id(tx);
+        tx.m_witness.resize(n_outputs);
+        const privkey_t seckey = m_keys.at(pubkey);
+
+        // Sign each output
+        for(size_t i = 0; i < n_outputs; i++) {
```

Comment:
This signing code seems duplicated with the code in `wallet::sign`, can we refactor it?

Reply:
Some of the code is duplicate.  As I was touching a lot of the code base, I didn't want to be too intrusive

Reply:
I think it's best to refactor the duplicate code in this PR. That will make the code more maintainable in the long run. You can split the refactors into separate commits to reduce the review burden.